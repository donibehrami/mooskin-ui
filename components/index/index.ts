import '../_utils/globals/global.css';

export { default as Alert, AlertCloseButton, AlertDescription, AlertIcon, AlertTitle } from '../Alert/Alert';
export { default as Skeleton, SkeletonCircle, SkeletonText } from '../Skeleton/Skeleton';
export { default as Drawer, DrawerBody, DrawerCloseButton, DrawerContent, DrawerFooter, DrawerHeader, DrawerOverlay } from '../Drawer/Drawer';
export { default as Stack, HStack, VStack } from '../Stack/Stack';
export { default as Box } from '../Box/Box';
export { default as Carousel } from '../Carousel/Carousel';
export { default as Checkbox, CheckboxDescription, CheckboxIcon, CheckboxLabel } from '../Checkbox/Checkbox';
export { default as DatePicker } from '../DatePicker/DatePicker';
export { default as DateRange } from '../DateRange/DateRange';
export { default as DateSelect } from '../DateSelect/DateSelect';
export { default as Grid, Row, Col } from '../Grid/Grid';
export {
    default as Input,
    InputContainer,
    InputDescription,
    InputEmoji,
    InputIcon,
    InputLabel,
    InputOption,
    InputOptionList,
    InputOptionListTitle,
    InputOverlay
} from '../Input/Input';
export { default as List, ListItem, ListItemBody, ListItemEnd, ListItemHead } from '../List/List';
export { default as Modal, ModalBody, ModalCloseButton, ModalContent, ModalFooter, ModalHeader, ModalOverlay } from '../Modal/Modal';
export { default as Pagination, PaginationButton } from '../Pagination/Pagination';
export { default as Radio, RadioDescription, RadioIcon, RadioLabel } from '../Radio/Radio';
export { default as Sidemenu, SidemenuItem } from '../Sidemenu/Sidemenu';
export { default as IconButton } from '../IconButton/IconButton';
export {
    default as Select,
    SelectContainer,
    SelectDescription,
    SelectFilter,
    SelectIcon,
    SelectLabel,
    SelectLoader,
    SelectOption,
    SelectOptionList,
    SelectOverlay,
    SelectPagination,
    SelectPlaceholder
} from '../Select/Select';
export { default as Switch } from '../Switch/Switch';
export { default as TextArea, TextareaDescription, TextareaLabel } from '../TextArea/TextArea';
export { default as Tabs, TabContent, TabHeader, Tab } from '../Tabs/Tabs';
export { default as Tags, TagClose, TagInput, Tag } from '../Tags/Tags';
export { default as Table, TableHeader, TableHeaderItem, TableRow, TableRowItem } from '../Table/Table';

export { default as IInputCallbackData } from '../_utils';
