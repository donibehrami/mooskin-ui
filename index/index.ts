// Context
export { MooskinContext, MooskinContextProvider, useMooskinContext, withMooskinContext } from '../components/Styled/MooskinContextProvider';

// Components
export { ActionsDropdown, ActionsDropdownArrow, ActionsDropdownItem } from '../components/ActionsDropdown/ActionsDropdown';
export { Alert, AlertCloseButton, AlertDescription, AlertIcon, AlertTitle } from '../components/Alert/Alert';
export { Button } from '../components/Button/Button';
export { Anchor } from '../components/Anchor/Anchor';
export { Skeleton, SkeletonCircle, SkeletonText } from '../components/Skeleton/Skeleton';
export {
    Drawer,
    DrawerBody,
    DrawerCloseButton,
    DrawerContent,
    DrawerFooter,
    DrawerHeader,
    DrawerOverlay
} from '../components/Drawer/Drawer';
export { Stack, HStack, VStack } from '../components/Stack/Stack';
export { Box } from '../components/Box/Box';
export { Carousel } from '../components/Carousel/Carousel';
export { Checkbox, CheckboxDescription, CheckboxIcon, CheckboxLabel } from '../components/Checkbox/Checkbox';
export { DateTimePicker } from '../components/DateTimePicker/DateTimePicker';
export { DatePicker } from '../components/DatePicker/DatePicker';
export { DateRange } from '../components/DateRange/DateRange';
export { DateSelect } from '../components/DateSelect/DateSelect';
export { Description } from '../components/Description/Description';
export { Layout } from '../components/Layout/Layout';
export { Selector, SelectorItem } from '../components/Selector/Selector';
export { Label } from '../components/Label/Label';
export {
    Expandable,
    ExpandableItem,
    ExpandableItemButton,
    ExpandableItemContainer,
    ExpandableItemContent,
    ExpandableItemText
} from '../components/Expandable/Expandable';
export { Grid, Row, Col } from '../components/Grid/Grid';
export {
    Input,
    InputContainer,
    InputEmoji,
    InputIcon,
    InputOption,
    InputOptionList,
    InputOptionListTitle,
    InputOverlay
} from '../components/Input/Input';
export { List, ListItem, ListItemBody, ListItemEnd, ListItemHead } from '../components/List/List';
export { Modal, ModalBody, ModalCloseButton, ModalContent, ModalFooter, ModalHeader, ModalOverlay } from '../components/Modal/Modal';
export { Pagination, PaginationButton } from '../components/Pagination/Pagination';
export { Radio, RadioDescription, RadioIcon, RadioLabel } from '../components/Radio/Radio';
export { Sidemenu, SidemenuItem } from '../components/Sidemenu/Sidemenu';
export { IconButton } from '../components/IconButton/IconButton';
export { Steps, StepContent, StepHeader, Step } from '../components/Steps/Steps';
export {
    Select,
    SelectContainer,
    SelectFilter,
    SelectIcon,
    SelectLoader,
    SelectOption,
    SelectOptionList,
    SelectOverlay,
    SelectPagination,
    SelectPlaceholder
} from '../components/Select/Select';
export { Switch, SwitchHandle } from '../components/Switch/Switch';
export { Loader } from '../components/Loader/Loader';
export { NumberLabel } from '../components/NumberLabel/NumberLabel';
export { LoadingBar } from '../components/LoadingBar/LoadingBar';
export { TextArea } from '../components/TextArea/TextArea';
export { TextEditor } from '../components/TextEditor/TextEditor';
export { Tabs, TabContent, TabHeader, Tab } from '../components/Tabs/Tabs';
export { Tags, TagClose, TagInput, Tag } from '../components/Tags/Tags';
export { Table, TableHeader, TableHeaderItem, TableRow, TableRowItem } from '../components/Table/Table';
export { Footer, FooterBody, FooterEnd, FooterHead } from '../components/Footer/Footer';

// Models
export {
    IActionsDropdownArrowComponentProps,
    IActionsDropdownComponentProps,
    IActionsDropdownItemComponentProps
} from '../components/ActionsDropdown/model';
export { IButtonComponentProps } from '../components/Button/model';
export { IBaseAlertComponentProps } from '../components/Alert/model';
export { IAnchorComponentProps } from '../components/Anchor/model';
export {
    IBaseBoxComponentProps,
    IBoxComponentProps,
    IAnchorBoxComponentProps,
    IBoxLabelComponentProps,
    IButtonBoxComponentProps,
    IInputBoxComponentProps,
    ITextAreaBoxComponentProps
} from '../components/Box/model';
export { ICarouselComponentProps } from '../components/Carousel/model';
export { ICheckboxComponentProps, ICheckboxIconComponentProps } from '../components/Checkbox/model';
export {
    IDatePickerCommonProps,
    IDatePickerComponentProps,
    IDatePickerKeyboardComponentProps,
    PickerType as DatePickerType
} from '../components/DatePicker/model';
export {
    IDateTimePickerCommonProps,
    IDateTimePickerComponentProps,
    IDateTimePickerKeyboardComponentProps,
    PickerType as DateTimePickerType
} from '../components/DateTimePicker/model';
export { IDateRangePickerComponentProps, IRangeSelection } from '../components/DateRange/model';
export { IDateSelectComponentProps } from '../components/DateSelect/model';
export { IDescriptionComponentProps } from '../components/Description/model';
export { ILabelComponentProps } from '../components/Label/model';
export { IDrawerComponentProps, IDrawerContentComponentProps, IDrawerOverlayComponentProps } from '../components/Drawer/model';
export { IExpandableCommonComponentProps, IExpandableComponentProps, IExpandableItemComponentProps } from '../components/Expandable/model';
export { IColProps, IGridProps, IRowProps } from '../components/Grid/model';
export { IIconButtonComponentProps } from '../components/IconButton/model';
export {
    IInputComponentProps,
    IInputContainerComponentProps,
    IInputEmojiComponentProps,
    IInputListComponentProps,
    IInputOptionComponentProps
} from '../components/Input/model';
export { ILayoutComponentProps } from '../components/Layout/model';
export { ILoaderComponentProps } from '../components/Loader/model';
export { ILoadingBarComponentProps } from '../components/LoadingBar/model';
export { IModalComponentProps, IModalContentComponentProps, IModalOverlayComponentProps } from '../components/Modal/model';
export { INumberLabelComponentProps } from '../components/NumberLabel/model';
export { IPaginationButtonComponentProps, IPaginationComponentProps } from '../components/Pagination/model';
export { IRadioComponentProps, IRadioIconComponentProps } from '../components/Radio/model';
export {
    ISelectComponentProps,
    ISelectFilterComponentProps,
    ISelectOptionComponentProps,
    ISelectPaginationComponentProps
} from '../components/Select/model';
export { ISelectorComponentProps, ISelectorItemComponentProps } from '../components/Selector/model';
export { ISidemenuComponentProps, ISidemenuItemComponentProps } from '../components/Sidemenu/model';
export {
    ICommonSkeletonComponentProps,
    ISkeletonCircleComponentProps,
    ISkeletonComponentProps,
    ISkeletonTextComponentProps
} from '../components/Skeleton/model';
export { IStackComponentProps } from '../components/Stack/model';
export { IStepCommonComponentProps, IStepComponentProps, IStepHeaderComponentProps, IStepsComponentProps } from '../components/Steps/model';
export { ISwitchComponentProps, ISwitchHandleComponentProps } from '../components/Switch/model';
export { ITabCommonComponentProps, ITabComponentProps } from '../components/Tabs/model';
export { ITagComponentProps, ITagsComponentProps, ITagsInputComponentProps } from '../components/Tags/model';
export { ITextAreaComponentProps } from '../components/TextArea/model';
export { IPersonalizationTag, ITextEditorComponentProps } from '../components/TextEditor/model';

// Global Styles
export { GlobalStyle } from '../components/Styled/GlobalStyles';

export { default as IInputCallbackData } from '../components/_utils';
