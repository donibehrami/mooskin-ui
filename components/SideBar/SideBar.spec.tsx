import * as React from 'react';
import {Item, SideBar, SmallIconButton} from '../index';

import { mount, shallow } from 'enzyme';

describe('SideBar', () => {

    test('renders SideBar correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <SideBar
                onClick={func}
                className="myClass"
                style={{color: 'blue'}}
                display
                button
            />
        );
        expect(tree).toMatchSnapshot();
    });

    test('renders Item correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <Item
                onClick={func}
                className="myClass"
                style={{color: 'blue'}}
                active
                href="www.moosend.com"
                label="Item"
                image="imagePath"
                onMouseEnter={func}
                onMouseLeave={func}
            />
        );
        expect(tree).toMatchSnapshot();
    });

    test('renders SideBar with Items correctly', () => {
        const func = jest.fn();

        const tree = mount(
            <SideBar
                onClick={func}
                className="myClass"
                style={{color: 'blue'}}
                display
                button
            >
                <Item
                    onClick={func}
                    className="myClass"
                    style={{color: 'blue'}}
                    active
                    href="www.moosend.com"
                    label="Item"
                    image="imagePath"
                    onMouseEnter={func}
                    onMouseLeave={func}
                />
                <Item
                    onClick={func}
                    className="myClass"
                    style={{color: 'blue'}}
                    active
                    href="www.moosend.com"
                    label="Item"
                    image="imagePath"
                    onMouseEnter={func}
                    onMouseLeave={func}
                />
            </SideBar>
        );
        expect(tree).toMatchSnapshot();
    });

    test('renders SideBar with Items and secondary SideBar correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <SideBar
                onClick={func}
                className="myClass"
                style={{color: 'blue'}}
                display
                button
            >
                <Item
                    onClick={func}
                    className="myClass"
                    style={{color: 'blue'}}
                    active
                    href="www.moosend.com"
                    label="Item"
                    image="imagePath"
                    onMouseEnter={func}
                    onMouseLeave={func}
                />
                <Item
                    onClick={func}
                    className="myClass"
                    style={{color: 'blue'}}
                    active
                    href="www.moosend.com"
                    label="Item"
                    image="imagePath"
                    onMouseEnter={func}
                    onMouseLeave={func}
                >
                    <SideBar display>
                        <Item
                            onClick={func}
                            className="myClass"
                            style={{color: 'blue'}}
                            active
                            href="www.moosend.com"
                            label="Item"
                            image="imagePath"
                            onMouseEnter={func}
                            onMouseLeave={func}
                        />
                    </SideBar>
                </Item>
            </SideBar>
        );
        expect(tree).toMatchSnapshot();
    });

    test('onClick callback function is called when sidebar button is clicked', () => {

        const func = jest.fn();

        const component = shallow(
            <SideBar display button onClick={func} />
        );

        component.find(SmallIconButton).simulate('click');
        expect(func).toHaveBeenCalled();
    });

    test('onMouseEnter callback function is called when mouse enters Item', () => {

        const func = jest.fn();
        const func2 = jest.fn();

        const component = shallow(
            <SideBar display>
                <Item onMouseEnter={func} onMouseLeave={func2}/>
            </SideBar>
        );

        expect(component.find(SmallIconButton)).toBeFalsy;

        component.find(Item).simulate('mouseenter');
        expect(func).toHaveBeenCalled();

        component.find(Item).simulate('mouseleave');
        expect(func2).toHaveBeenCalled();
    });

    test('SideBar prop tests', () => {

        const component = shallow(
            <SideBar
                className="myClass"
                style={{color: 'blue'}}
                display
                button
            >
                <Item
                    className="myClass"
                    style={{color: 'blue'}}
                    active
                    href="www.moosend.com"
                    label="Item"
                    image="imagePath"
                />
                <Item
                    className="myClass"
                    style={{color: 'blue'}}
                    active
                    href="www.moosend.com"
                    label="Item"
                    image="imagePath"
                />
            </SideBar>
        );

        expect(component.find(SmallIconButton)).toBeTruthy;
        expect(component.find('.sidebar').prop('className')).toContain('myClass');
        expect(component.find('.sidebar').prop('style')).toEqual({color: 'blue'});

        expect(component.find(Item).length).toEqual(2);
        expect(component.find(Item).first().prop('className')).toEqual('myClass');
        expect(component.find(Item).first().prop('style')).toEqual({color: 'blue'});
        expect(component.find(Item).first().prop('href')).toEqual('www.moosend.com');
        expect(component.find(Item).first().prop('image')).toEqual('imagePath');
    });
});
