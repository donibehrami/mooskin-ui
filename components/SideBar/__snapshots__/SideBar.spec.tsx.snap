// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders Item correctly 1`] = `
<div
  className="item-component itemContainer myClass"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "color": "blue",
    }
  }
>
  <div
    className="arrow arrowImg"
  />
  <a
    className="anchor activeItem"
    href="www.moosend.com"
  >
    <img
      className="image"
      src="imagePath"
    />
    <span>
      Item
    </span>
  </a>
  <div />
</div>
`;

exports[`SideBar renders SideBar correctly 1`] = `
<div
  className="sidebar-component"
>
  <div>
    <SmallIconButton
      className="button"
      icon="view headline"
      onClick={[Function]}
      style={Object {}}
    />
  </div>
  <div
    className="sidebar myClass sidebarOn"
    style={
      Object {
        "color": "blue",
      }
    }
  />
  <div
    className="cover"
    onClick={[Function]}
  />
</div>
`;

exports[`SideBar renders SideBar with Items and secondary SideBar correctly 1`] = `
<div
  className="sidebar-component"
>
  <div>
    <SmallIconButton
      className="button"
      icon="view headline"
      onClick={[Function]}
      style={Object {}}
    />
  </div>
  <div
    className="sidebar myClass sidebarOn"
    style={
      Object {
        "color": "blue",
      }
    }
  >
    <Item
      active={true}
      className="myClass"
      href="www.moosend.com"
      image="imagePath"
      label="Item"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "blue",
        }
      }
    />
    <Item
      active={false}
      className="myClass"
      href="www.moosend.com"
      image="imagePath"
      label="Item"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "blue",
        }
      }
    >
      <SideBar
        className=""
        display={true}
        style={Object {}}
      >
        <Item
          active={true}
          className="myClass"
          href="www.moosend.com"
          image="imagePath"
          label="Item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "blue",
            }
          }
        />
      </SideBar>
    </Item>
  </div>
  <div
    className="cover"
    onClick={[Function]}
  />
</div>
`;

exports[`SideBar renders SideBar with Items correctly 1`] = `
<SideBar
  button={true}
  className="myClass"
  display={true}
  onClick={[Function]}
  style={
    Object {
      "color": "blue",
    }
  }
>
  <div
    className="sidebar-component"
  >
    <div>
      <SmallIconButton
        className="button"
        icon="view headline"
        onClick={[Function]}
        style={Object {}}
      >
        <button
          className="button-icon-component
                                 button
                                 normalButton
                                 
                                 button"
          onClick={[Function]}
          style={Object {}}
        >
          <i
            className="material-icons icon"
          >
            view_headline
          </i>
        </button>
      </SmallIconButton>
    </div>
    <div
      className="sidebar myClass sidebarOn"
      style={
        Object {
          "color": "blue",
        }
      }
    >
      <Item
        active={true}
        className="myClass"
        href="www.moosend.com"
        image="imagePath"
        label="Item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        <div
          className="item-component itemContainer myClass"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "blue",
            }
          }
        >
          <div
            className="arrow arrowImg"
          />
          <a
            className="anchor activeItem"
            href="www.moosend.com"
          >
            <img
              className="image"
              src="imagePath"
            />
            <span>
              Item
            </span>
          </a>
          <div />
        </div>
      </Item>
      <Item
        active={false}
        className="myClass"
        href="www.moosend.com"
        image="imagePath"
        label="Item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        <div
          className="item-component itemContainer myClass"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "blue",
            }
          }
        >
          <div
            className="arrow arrowImg"
          />
          <a
            className="anchor "
            href="www.moosend.com"
          >
            <img
              className="image"
              src="imagePath"
            />
            <span>
              Item
            </span>
          </a>
          <div />
        </div>
      </Item>
    </div>
    <div
      className="cover"
      onClick={[Function]}
    />
  </div>
</SideBar>
`;
